### ISSUE TITLE
Configuring version may have no effect

### ISSUE THREAD

**Issue by rkrisztian** [bonigarcia/webdrivermanager#232]
Provided I do the following first on a fresh system that downloads version `0.21.0` of geckodriver:

```java
WebDriverManager.firefoxdriver().setup();
```

In my experience, when I do this:

```java
WebDriverManager.firefoxdriver().version("0.20.1").setup();
```

Then still version `0.21.0` is used. This is counter-intuitive, because all I wanted was to use that particular version.

If I change the code to:

```java
WebDriverManager.firefoxdriver().version("0.20.1").forceDownload().setup();
```

Then the right version will be used, however, I will lose the feature of using cached downloads when possible. I would rather have the cache working too.

I find this problem rather important, please help.

**Update:** Now I can't reproduce this bug when executing this code:

```
WebDriverManager.firefoxdriver().version("0.19.0").setup();
```

However, if I delete everything from `~/.m2/repository/webdriver/geckodriver/linux64/`, then I can reproduce the original problem again.

The documentation seem very weird about `forceDownload()` and `forceCache()`, also a bit contradictory. Seemingly the behavior I expect does not seem to be supported, i.e. being able to use whatever version I want, and that the specified exact version would be downloaded unless it is already there in the cache. I don't understand why WebDriverManager doesn't work like this, so frustrating.

**Comment by rkrisztian** [comment]
The bug seem to appear only if there is only 1 geckodriver in the cache.

The following workaround gave me some relief, even though it's a bit ugly:

```java
		String version = "0.18.0"; // I tried "0.21.0" first with an empty cache.
		WebDriverManager.firefoxdriver().version(version).setup();
		String expectedPath = System.getProperty(GeckoDriverService.GECKO_DRIVER_EXE_PROPERTY).replaceAll("(\\d+\\.)+\\d+",
				version);
		if (!new File(expectedPath).exists()) {
			WebDriverManager.firefoxdriver().forceDownload().version(version).setup();
		}
```

**Comment by bonigarcia** [comment]
In fact this a bug, thanks for reporting.

It was already solved in the master branch, nevertheless it is not released yet. It will be soon.

**Comment by bonigarcia** [comment]
This should be fixed in version 2.2.5
